utils/build_swift/build_swift/driver_arguments.py

```diff
@@ -568,6 +568,10 @@ def create_argument_parser():
                 'library dependencies of the corelibs and other Swift repos, '
                 'such as the libcurl dependency of FoundationNetworking')
 
+    option('--disable-append-cross-compile-hostname-to-destdir', toggle_true,
+           help="Turn off the default of appending each cross-compilation hostname "
```

Comment:
I would suggest renaming this options, to better highlight it controls cross-compilation, to match the style of similar options, and to ease readability of the condition in `build-script-impl` to an extent

```suggestion
    option('--cross-compile-append-host-target-to-destdir', toggle_true,
           default=True,
           help="Turn off the default of appending each cross-compilation host target "
```

This will require renaming in the other files and adjusting accordingly the conditions and the defaults

Reply:
(optional) I would expand the help to describe a bit the scenario this flag is meant for (e.g. "this could be useful when cross compiling for a single target and we are not interested in targeting the architecture of the build machine")

Reply:
Sure.